content/kb/tutorials/deploy/kubernetes.md

```diff
@@ -0,0 +1,288 @@
+---
+title: Deploy Streamlit using Kubernetes
+slug: /knowledge-base/tutorials/deploy/kubernetes
+---
+
+# Deploy Streamlit using Kubernetes
+
+## Introduction
+
+So you have an amazing app and you want to start sharing it with other people, what do you do? You have a few options. First, where do you want to run your Streamlit app, and how do you want to access it?
+
+- **On your corporate network** - Most corporate networks are closed to the outside world. You typically use a VPN to log onto your corporate network and access resources there. You could run your Streamlit app on a server in your corporate network for security reasons, to ensure that only folks internal to your company can access it.
+- **On the cloud** - If you'd like to access your Streamlit app from outside of a corporate network, or share your app with folks outside of your home network or laptop, you might choose this option. In this case, it'll depend on your hosting provider. We have community-submitted guides from Heroku, AWS, and other providers [here](/knowledge-base/deploy/deploy-streamlit-heroku-aws-google-cloud).
+
+Wherever you decide to deploy your app, you will first need to containerize it. This guide walks you through using Kubernetes to deploy your app. If you prefer Docker see [Deploy Streamlit using Docker](/knowledge-base/tutorials/deploy/docker).
+
+## Prerequisites
+
+1. [Install Docker Engine](#install-docker-engine)
+2. [Install the gcloud CLI](#install-the-gcloud-cli)
+
+### Install Docker Engine
+
+If you haven't already done so, install [Docker](https://docs.docker.com/engine/install/#server) on your server. Docker provides `.deb` and `.rpm` packages from many Linux distributions, including:
+
+- [Debian](https://docs.docker.com/engine/install/debian/)
+- [Ubuntu](https://docs.docker.com/engine/install/ubuntu/)
+
+Verify that Docker Engine is installed correctly by running the `hello-world` Docker image:
+
+```bash
+sudo docker run hello-world
+```
+
+<Tip>
+
+Follow Docker's official [post-installation steps for Linux](https://docs.docker.com/engine/install/linux-postinstall/) to run Docker as a non-root user, so that you don't have to preface the `docker` command with `sudo`.
+
+</Tip>
+
+### Install the gcloud CLI
+
+In this guide, we will orchestrate Docker containers with Kubernetes and host docker images on the Google Container Registry (GCR). As GCR is a Google-supported Docker registry, we need to register [`gcloud`](https://cloud.google.com/sdk/gcloud/reference) as the Docker credential helper.
+
+Follow the official documentation to [Install the gcloud CLI](https://cloud.google.com/sdk/docs/install) and initialize it.
```

Comment:
The easiest way to install gcloud is to run one command:
```
curl https://sdk.cloud.google.com/ | bash -
```
What do you think to add this?

Reply:
Is there a reason the one liner is omitted from the official [Install instructions](https://cloud.google.com/sdk/docs/install#linux)? i.e. are there potential edge cases that might occur, which can be avoided by following the official instructions?

Reply:
In general I think we should link to the official instructions, rather than maintaining a parallel set of directions. The more custom instructions we have, the more we have to maintain.

Reply:
Here is guide about this short installation method: https://cloud.google.com/sdk/docs/downloads-interactive